crates/rattler_config/src/lib.rs

```diff
@@ -0,0 +1,2607 @@
+
+    /// Dependency Pinning strategy used for dependency modification through
+    /// automated logic like `pixi add`
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub pinning_strategy: Option<PinningStrategy>,
+
+    #[serde(skip)]
+    #[serde(alias = "loaded_from")] // BREAK: remove to stop supporting snake_case alias
+    pub loaded_from: Vec<PathBuf>,
+
+    #[serde(skip, default = "default_channel_config")]
+    channel_config: ChannelConfig,
+
+    /// Configuration for repodata fetching.
+    #[serde(alias = "repodata_config")] // BREAK: remove to stop supporting snake_case alias
+    #[serde(default, skip_serializing_if = "RepodataConfig::is_empty")]
+    pub repodata_config: RepodataConfig,
+
+    /// Configuration for PyPI packages.
+    #[serde(default)]
+    #[serde(skip_serializing_if = "PyPIConfig::is_default")]
+    pub pypi_config: PyPIConfig,
+
+    /// Configuration for S3.
+    #[serde(default)]
+    #[serde(skip_serializing_if = "HashMap::is_empty")]
+    pub s3_options: HashMap<String, S3Options>,
+
+    /// The option to specify the directory where detached environments are
+    /// stored. When using 'true', it defaults to the cache directory.
+    /// When using a path, it uses the specified path.
+    /// When using 'false', it disables detached environments, meaning it moves
+    /// it back to the .pixi folder.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub detached_environments: Option<DetachedEnvironments>,
+
+    /// Shell-specific configuration
+    #[serde(default)]
+    #[serde(skip_serializing_if = "ShellConfig::is_default")]
+    pub shell: ShellConfig,
+
+    /// Experimental features that can be enabled.
+    #[serde(default)]
+    #[serde(skip_serializing_if = "ExperimentalConfig::is_default")]
+    pub experimental: ExperimentalConfig,
+
+    /// Concurrency configuration for pixi
+    #[serde(default)]
+    #[serde(skip_serializing_if = "ConcurrencyConfig::is_default")]
+    pub concurrency: ConcurrencyConfig,
+
+    /// Run the post link scripts
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub run_post_link_scripts: Option<RunPostLinkScripts>,
+
+    /// Https/Http proxy configuration for pixi
+    #[serde(default)]
+    #[serde(skip_serializing_if = "ProxyConfig::is_default")]
+    pub proxy_config: ProxyConfig,
+
+    /// Build configuration for pixi and rattler-build
+    #[serde(default)]
+    #[serde(skip_serializing_if = "BuildConfig::is_default")]
+    pub build: BuildConfig,
+
+    //////////////////////
+    // Deprecated fields //
+    //////////////////////
+    #[serde(default)]
+    #[serde(alias = "change_ps1")] // BREAK: remove to stop supporting snake_case alias
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub change_ps1: Option<bool>,
+
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub force_activate: Option<bool>,
+}
+
+impl Default for Config {
```

Comment:
couldn't we just derive default here if we implement `Default` for `ChannelConfig`?